### ISSUE TITLE
FAILED NIGHTLY: Kokkos_ENABLE_PTHREAD does not add -lpthread 

### ISSUE THREAD

**Issue by jeffmiles63** [kokkos/kokkos#2363]
need something like this in kokkos_arch.cmake

IF(KOKKOS_ENABLE_PTHREAD)
  SET(PTHREAD_LINK_OPTIONS)
  LIST(APPEND PTHREAD_LINK_OPTIONS ${KOKKOS_LINK_OPTIONS})
  LIST(APPEND PTHREAD_LINK_OPTIONS -lpthread)
  GLOBAL_SET(KOKKOS_LINK_OPTIONS ${PTHREAD_LINK_OPTIONS})
ENDIF()

**Comment by jjwilke** [comment]
I have heard horror stories with using `find_package(Threads)` in CMake. I don't have a ton of experience with this.  Probably we need:

https://cmake.org/cmake/help/v3.2/module/FindThreads.html

and then define
````
CMAKE_THREAD_PREFER_PTHREAD
````
and also
````
THREADS_PREFER_PTHREAD_FLAG
````

**Comment by jjwilke** [comment]
Do we want this to be `-lpthread` or `-pthread`?

**Comment by jeffmiles63** [comment]
I think we actually want -pthread which is both a compile and link option...will have to test to make sure.

-pthread is more encompassing and it is more portable.

**Comment by jjwilke** [comment]
Yeah, that's what I was thinking as well. I believe we need to add:

````
SET(THREADS_PREFER_PTHREAD_FLAG TRUE)
find_package(Threads REQUIRED)
....
target_link_libraries(kokkoscore PUBLIC Threads::Threads)
````

I think we also need to add the `KokkosConfig.in` the import again:

````
SET(THREADS_PREFER_PTHREAD_FLAG TRUE)
find_package(Threads REQUIRED)
````

**Comment by masterleinad** [comment]
Fixed by #2467.
